### PR TITLE
Minio no shadow variable

### PR DESCRIPTION
**Pull request by dholth**
Update test fixture to not shadow variable.

Update CHANGELOG.md to reflect that bugfix only affects test suite.

### CODE REVIEW

File: conda/testing/gateways/fixtures.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 MINIO_EXE = which("minio")
 
 
-def minio_s3_server(xprocess, tmp_path):
+def minio_s3_server_runner(xprocess, tmp_path):
```
[the remark this answers is not in the document]
Comment (dholth) [reply conda/testing/gateways/fixtures.py:18]: I thought the original author wanted the function for debugging.

I suppose pytest is smart enough to only create the fixture on demand.

[the remark this answers is not in the document]
Comment (dholth) [reply conda/testing/gateways/fixtures.py]: Done

File: conda/testing/gateways/fixtures.py [hunk 2]
```diff
@@ -15,6 +15,9 @@
 MINIO_EXE = which("minio")
 
 
+# rely on tests not requesting this fixture, and pytest not creating this if
+# MINIO_EXE was not found
+pytest.fixture()
```
Comment (kenodegard) [inline conda/testing/gateways/fixtures.py:20]:
```suggestion
@pytest.fixture()
```

**Comment by kenodegard** [comment]
pre-commit.ci autofix

**Review by kenodegard — APPROVED** [review]

**Pull request merged**